app/src/main/java/com/google/samples/apps/sunflower/compose/gallery/GalleryScreen.kt

```diff
@@ -59,41 +70,85 @@ fun GalleryScreen(
         plantPictures = viewModel.plantPictures,
         onPhotoClick = onPhotoClick,
         onUpClick = onUpClick,
+        onPullToRefresh = viewModel::refreshData,
     )
 }
-
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 private fun GalleryScreen(
     plantPictures: Flow<PagingData<UnsplashPhoto>>,
     onPhotoClick: (UnsplashPhoto) -> Unit = {},
     onUpClick: () -> Unit = {},
+    onPullToRefresh: () -> Unit,
 ) {
     Scaffold(
         topBar = {
             GalleryTopBar(onUpClick = onUpClick)
         },
     ) { padding ->
-        val pagingItems: LazyPagingItems<UnsplashPhoto> = plantPictures.collectAsLazyPagingItems()
-        LazyVerticalGrid(
-            columns = GridCells.Fixed(2),
-            modifier = Modifier.padding(padding),
-            contentPadding = PaddingValues(all = dimensionResource(id = R.dimen.card_side_margin))
-        ) {
-            // TODO update this implementation once paging Compose supports LazyGridScope
-            // See: https://issuetracker.google.com/issues/178087310
-            items(
-                count = pagingItems.itemCount,
-                key = { index ->
-                    val photo = pagingItems[index]
-                    "${ photo?.id ?: ""}${index}"
+
+        var isRefreshing by remember { mutableStateOf(false) }
```

Comment:
Extra `isRefreshing` state is not needed. `pullToRefreshState` should be the source of truth instead of having 2 different "is refreshing" states.